migrate.py

```diff
@@ -389,150 +400,151 @@ def assemble_collections(spec, args):
 
     seen = {}
     migrated_to_collection = defaultdict(set)
-    for collection in spec.keys():
+    for namespace in spec.keys():
```

Comment:
Let's try to avoid shifting the indentation in order to maintain 
```suggestion
    for namespace, collection in (ns, coll for ns, colls in spec.items() for coll in colls.keys()):
```

Reply:
i find that the indentation helps define scope also, not sure if its worth it to avoid 1 merge conflict

Reply:
There are no branches in outer loops, it doesn't make sense to do this. The only important context is in the internal loop. Also,  you can put this into a variable using generator expressions to make it more clear.

Reply:
Also, it's hard to read large diffs produced by these many shifts.

Reply:
that is the thing, we'll need to add branches at namespace level to deal with differences at that level, so no sense in collapsing now and having to expand later

Reply:
It makes sense to expand it separately when needed. Now it just creates more problems

Reply:
Just spent two hours rebasing #39 because if this. Not cool.